src/middlewared/middlewared/plugins/filesystem_/perm_check.py

```diff
@@ -1,10 +1,32 @@
 import errno
+import os
 import pathlib
 
+from multiprocessing import Pipe, Process
+from multiprocessing.connection import Connection
+
 from middlewared.schema import accepts, Bool, Dict, returns, Str
 from middlewared.service import CallError, Service
 
-from middlewared.utils.osc import run_command_with_user_context
+from middlewared.utils.osc import set_user_context
+
+
+def check_perms(user: str, path: str, check_perms: dict, pipe: Connection) -> None:
```

Comment:
Maybe check_access() rather than check_perms() since former is somewhat more like what we're doing.